src/elf/dyn.rs

```diff
@@ -466,6 +472,7 @@ macro_rules! elf_dyn_std_impl {
             pub fn from_fd(mut fd: &File, phdrs: &[$phdr]) -> Result<Option<Vec<Dyn>>> {
                 for phdr in phdrs {
                     if phdr.p_type == PT_DYNAMIC {
+                        // FIXME: validate filesz before allocating
```

Comment:
I don't know how to validate this. Probably we could do `fd.take(filesz).read_to_end()` (and then check the correct length was read?), but then need to convert that `Vec[u8]` into a `Vec[Dyn]`.

Reply:
I think this method is only used by things that load off disk like a dynamic linker so it shouldn’t be on the parser code path; however yes we should do something here but I wouldn’t worry about it for now